src/Insights.php

```diff
@@ -104,8 +104,8 @@ public function add_extra( $data = array() ) {
      *
      * @return \self
      */
-    public function notice( $text ) {
-        $this->notice = $text;
+    public function notice(string $text='' ) {
+        $this->notice = sanitize_text_field($text);
```

Comment:
It's not necessary to sanitize text here as it's not coming from the general untrusted users. Only when we are outputting, we need to escape the output with [escape functions](https://developer.wordpress.org/themes/theme-security/data-sanitization-escaping/#escaping-securing-output).